cpp/src/arrow/filesystem/hdfs_internal.h

```diff
@@ -33,8 +38,13 @@ namespace arrow {
 
 class Status;
 
+using internal::IOErrorFromErrno;
+
 namespace io::internal {
 
+class HdfsReadableFile;
+class HdfsOutputStream;
+
```

Comment:
Why are we using the `io::internal` namespace here instead of `fs::internal`?

Reply:
I think I wanted to change as little as possible. But yeah, should be changed to `fs::internal`. Will do!